src-electron/db/db-api.js

```diff
@@ -482,9 +490,18 @@ async function loadSchema(db, schemaPath, zapVersion, sqliteFile = null) {
       db = await initDatabase(sqliteFile)
     }
   }
+
   if (context.mustLoad) {
     await performSchemaLoad(db, context.data)
     await updateCurrentSchemaCrc(db, context)
+    for (let i = 0; i < rows.length; i++) {
+      await insertOrReplaceSetting(
+        db,
+        rows[i].CATEGORY,
```

Comment:
In principle this works, but it hardcodes the columns of the setting table here. Let's just clean this up a bit:

Preferred pattern is:
   0.) add the "selectSettings()", which will read the settings from the table and create an array of JS objects that have keys: "category", "key", "value", so add the mapping to that db-mapping.js. It should not do "SELECT *", but instead it should do "SELECT CATEGORY, KEY, VALUE" and then map the rows into the JS objects, like most of our query functions do.
   1.) Add the "updateSetting()" that receives either a single { category:, key:, value: } object, or an array of them, and performs the "INSERT OR UPDATE".
   2.) Then make the db-api.js do the same thing you've made it here, except call these methods to interact with the SETTING table.
   
Otherwise, once we clean this up, this is good to go I think.

Reply:
Okay thanks! I am on it